### PR TITLE
DSFAAP-511: Destination type page

### PR DESCRIPTION
**Pull request by joseluisgraa**

### CODE REVIEW

File: src/server/common/model/answer/destination-type.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    } else if (destinationType === 'dedicated-sale') {
+      return 'Dedicated sale for TB (orange market)'
+    } else if (destinationType === 'afu') {
+      return 'Approved finishing unit (AFU)'
+    } else if (destinationType === 'other') {
+      return 'Another destination'
+    } else {
+      return ''
+    }
+  }
```
Comment (hughfdjackson) [inline src/server/common/model/answer/destination-type.js:52]: No pressure to do this (or even do it now), but an object lookup can be nice here: 

```javascript
const destinationTypeMapping = { 
  'slaughter': 'Slaughter',
  'dedicated-sale': 'Dedicated sale for TB (orange market)',
   // etc
}

return destinationTypeMapping[destinationType] ?? ''
```

File: src/server/destination/exit-page/index.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { Page } from '../../common/model/page/page-model.js'
+
+export class DestinationExitPage extends Page {
+  urlPath = '/destination/can-not-use-service'
```
Comment (hughfdjackson) [inline src/server/destination/exit-page/index.js:4]: Is this right?  I was expecting 'cannot-user-service' (based on spelling)

File: src/server/task-list/controller.js [hunk 3]
```diff
@@ -32,9 +34,11 @@ export const taskListGetController = {
 
     const destinationGdsTask = buildGdsTaskItem({
       title: 'Movement destination',
-      initialLink: '#',
-      summaryLink: '#',
-      isValid: destination.validate().isValid,
+      initialLink:
+        destinationValidity.firstInvalidPage?.urlPath ??
+        destination.firstPage.urlPath,
```
Comment (hughfdjackson) [inline src/server/task-list/controller.js:39]: Hrm - I can see this replicates the pattern we've gone with above, but I *believe* this isn't the pattern we went for.  It was either the firstPage, or it was the summary page.

**Review by hughfdjackson — CHANGES_REQUESTED** [review]
The change requested is on the URL.  

I don't think the task logic is correct, but I don't expect that to be fixed in this PR.

File: src/server/destination/exit-page/index.js (hunk 2 above)
Comment (joseluisgraa) [reply src/server/destination/exit-page/index.js:4]: That is the URL in the prototype

File: src/server/common/model/answer/destination-type.js (hunk 1 above)
Comment (joseluisgraa) [reply src/server/common/model/answer/destination-type.js:52]: Nice one I'll update it

File: src/server/destination/exit-page/index.js (hunk 2 above)
Comment (hughfdjackson) [reply src/server/destination/exit-page/index.js:4]: okay, let's go with that for now.  It looks like we'll be doing a url sweep later.

**Review by hughfdjackson — APPROVED** [review]

File: user-journey-tests/specs/destination/destinationSelection.spec.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      destinationSelectionPage.pagePath,
+      destinationSelectionPage.pageTitle
+    )
+  })
+
+  it('Should verify that the destination page errors when no option is selected', async () => {
+    await destinationSelectionPage.destinationSelectionErrorTest()
+  })
+
+  it('Should select on the farm radio and continue', async () => {
```
Comment (joseluisgraa) [inline user-journey-tests/specs/destination/destinationSelection.spec.js:22]: on the farm?

**Pull request merged**